src/core/lib/iomgr/tcp_server_posix.c

```diff
@@ -736,18 +745,21 @@ void grpc_tcp_server_shutdown_starting_add(grpc_tcp_server *s,
 
 void grpc_tcp_server_unref(grpc_exec_ctx *exec_ctx, grpc_tcp_server *s) {
   if (gpr_unref(&s->refs)) {
-    /* Complete shutdown_starting work before destroying. */
     grpc_exec_ctx local_exec_ctx = GRPC_EXEC_CTX_INIT;
+    bool finish_ctx = false;
+    /* FIXME: API allows a NULL exec_ctx, although this might cause us to delete
```

Comment:
There were some test cases that passed NULL, but they are gone now, so I'll remove it from the API.

There is some vendor-specific code that passes NULL. Did you ever consider having exec_ctx come from thread-local data rather than a parameter to every function?